### PR TITLE
provide integration test support for registry 2.2

### PR DESCRIPTION
**Pull request by FelixFan1992**

### CODE REVIEW

File: integration-tests/contracts/contract_deployer.go [hunk 1]
```diff
@@ -1307,6 +1394,22 @@ func (e *EthereumContractDeployer) LoadKeeperRegistry(address common.Address, re
 			registry2_1: instance.(*iregistry21.IKeeperRegistryMaster),
 			version:     registryVersion,
 		}, err
+	case eth_contracts.RegistryVersion_2_2: // why the contract name is not the same as the actual contract name?
```
Comment (FelixFan1992) [inline integration-tests/contracts/contract_deployer.go:1397]: @anirudhwarrier do you know why?

File: integration-tests/contracts/ethereum_keeper_contracts.go [hunk 2]
```diff
@@ -248,9 +253,25 @@ func (rcs *KeeperRegistrySettings) EncodeOnChainConfig(registrar string, registr
 			UpkeepPrivilegeManager: registryOwnerAddress,
 		}
 
-		encodedOnchainConfig, err := utilsABI.Methods["_onChainConfig"].Inputs.Pack(&onchainConfigStruct)
+		encodedOnchainConfig, err := utilsABI21.Methods["_onChainConfig"].Inputs.Pack(&onchainConfigStruct)
 
 		return encodedOnchainConfig, err
+	} else if rcs.RegistryVersion == ethereum.RegistryVersion_2_2 {
+		chainModuleBaseAddr, _, _, err := client.DeployContract("ChainModuleBase", func(
```
Comment (anirudhwarrier) [inline integration-tests/contracts/ethereum_keeper_contracts.go:260]: @FelixFan1992 can we have `ChainModuleBase` contract deployment part inside contract_deployer.go, inside `DeployKeeperRegistry`? That will let us avoid passing the client to this function.

File: integration-tests/contracts/contract_deployer.go (hunk 1 above)
Comment (anirudhwarrier) [reply integration-tests/contracts/contract_deployer.go:1397]: No particular reason. The name here doesn't matter, only the name in the contract will show up on chain when deployed. Feel free to change the name here to keep it in sync with the contract name.

File: integration-tests/contracts/ethereum_keeper_contracts.go (hunk 2 above)
Comment (FelixFan1992) [reply integration-tests/contracts/ethereum_keeper_contracts.go]: fixed

File: integration-tests/contracts/contract_deployer.go [hunk 3]
```diff
@@ -1225,11 +1222,156 @@ func (e *EthereumContractDeployer) DeployKeeperRegistry(
...
+		) (common.Address, *types.Transaction, interface{}, error) {
+
+			return registrylogicb22.DeployAutomationRegistryLogicB(
+				auth,
+				backend,
+				common.HexToAddress(opts.LinkAddr),
+				common.HexToAddress(opts.ETHFeedAddr),
+				common.HexToAddress(opts.GasFeedAddr),
+				*automationForwarderLogicAddr,
+				common.HexToAddress("0x0000000000000000000000000000000000000000"),
```
Comment (anirudhwarrier) [inline integration-tests/contracts/contract_deployer.go:1283]: should be different for polygon zkEVM

**Review by anirudhwarrier — APPROVED** [review]
LGTM

**Review by jmank88 — APPROVED** [review]

**Review by chudilka1 — APPROVED** [review]

**Review by tateexon — APPROVED** [review]

**Pull request merged**